tensorboard/webapp/plugins/npmi/views/data_selection/metric_search/metric_search_container.ts

```diff
@@ -0,0 +1,111 @@
+/* Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+import {Component, ChangeDetectionStrategy} from '@angular/core';
+import {select, Store} from '@ngrx/store';
+
+import {map, combineLatest} from 'rxjs/operators';
+
+import {State} from '../../../../../app_state';
+import {
+  getMetricFilterKeys,
+  getMetricsRegex,
+  getRunToMetrics,
+} from '../../../store';
+import {getRunSelection} from '../../../../../core/store/core_selectors';
+import * as npmiActions from '../../../actions';
+
+/** @typehack */ import * as _typeHackRxjs from 'rxjs';
+
+@Component({
+  selector: 'npmi-metric-search',
+  template: `
+    <metric-search-component
+      [regexFilterValue]="metricsRegex$ | async"
+      [completions]="completions$ | async"
+      [isRegexFilterValid]="isMetricsFilterValid$ | async"
+      (onRegexFilterValueChange)="onFilterChange($event)"
+    ></metric-search-component>
+  `,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class MetricSearchContainer {
+  readonly metricsRegex$ = this.store.select(getMetricsRegex);
+  readonly activeRuns$ = this.store.pipe(select(getRunSelection)).pipe(
+    map((runs) => {
+      let activeRuns: string[] = [];
+      if (runs) {
+        for (let run of runs) {
+          if (run[1]) {
+            activeRuns.push(run[0]);
+          }
+        }
+      }
+      return activeRuns;
+    })
+  );
+  readonly getRunToMetrics$ = this.store.pipe(select(getRunToMetrics));
+  readonly completions$ = this.store
+    .pipe(select(getRunToMetrics))
+    .pipe(
+      combineLatest(this.activeRuns$),
```

Comment:
This pattern is new to me!  I've never seen a `.pipe(combineLatest)` used, since the rxjs docs example only uses combineLatest outside of the pipe.

If the intent of `completion$` is to possibly emit whenever any of these changes - runToMetrics, activeRun, metricFilterKeys, metricsRegex, then could we do a single `combineLatest([runToMetrics, activeRun, ...])` at the start?

I think it would be clearer to see when this observable could emit.

https://www.learnrxjs.io/learn-rxjs/operators/combination/combinelatest

Reply:
Done.